creator-node/src/tracer.ts

```diff
@@ -65,6 +85,79 @@ export const setupTracing = () => {
   provider.register()
 }
 
+/**
+ * Higher-order function that adds opentelemetry tracing to a function.
+ * This wrapper works for both sync and async functions
+ *
+ * @param {string?} param.name optional name to give to the span, defaults to the function name
+ * @param {TFunction} param.fn the generic function to instrument
+ * @param {SpanOptions} param.options objects to pass into the span
+ * @returns the instrumented function
+ * @throws rethrows any errors from the original fn
+ *
+ * Usage of this would look like
+ * ```
+ * const someFunction = instrumentTracing({ fn: _someFunction })
+ * const result = someFunction(args))
+ * // or
+ * const result = await someFunction(args)
+ * ```
+ */
+export const instrumentTracing = <TFunction extends (...args: any[]) => any>({
+  name,
+  fn,
+  options
+}: {
+  name?: string
+  fn: TFunction
+  options?: SpanOptions
+}) => {
+  // build a wrapper around `fn` that accepts the same parameters and returns the same return type
+  const wrapper = (...args: Parameters<TFunction>): ReturnType<TFunction> => {
+    const spanName = name || fn.name
+    const spanOptions = options || {}
+    return tracing
+      .getTracer()
+      .startActiveSpan(spanName, spanOptions, (span: Span) => {
+        try {
+          tracing.setSpanAttribute(tracing.CODE_FUNCTION, fn.name)
+
+          // TODO add skip parameter to instrument testing function to NOT log certain args
+          // tracing.setSpanAttribute('args', JSON.stringify(args))
+          const result = fn.call(this, ...args)
```

Comment:
since this is inside an arrow function, wouldn't the context of `this` not be preserved from the calling function?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#call_apply_and_bind

Reply:
I hate javascript lol - lemme test this one some code and see if I can tell a difference. I imagine I'll end up changing it to return `function () {}` instead of `() => {}`

Reply:
This is what I was basing it off anyway - https://blog.sentry.io/2016/02/03/wrap-javascript-functions

Reply:
yeah just gotta change from arrow to regular function. this is why i always prefer regular named functions honestly (hoisted, not anonymous, preserve this)